connectors/couchbase/translator-couchbase/src/main/java/org/teiid/translator/couchbase/CouchbaseMetadataProcessor.java

```diff
@@ -0,0 +1,718 @@
+        param.setAnnotation(CouchbasePlugin.Util.getString("deleteDocument.id.Annotation")); //$NON-NLS-1$
+        param.setNullType(No_Nulls);
+        param = metadataFactory.addProcedureParameter(KEYSPACE, TypeFacility.RUNTIME_NAMES.STRING, Type.In, deleteDocument); 
+        param.setAnnotation(CouchbasePlugin.Util.getString("deleteDocument.keyspace.Annotation")); //$NON-NLS-1$
+        param.setNullType(No_Nulls);
+        metadataFactory.addProcedureResultSetColumn(RESULT, TypeFacility.RUNTIME_NAMES.CLOB, deleteDocument); 
+        
+        Procedure getTextMetadataDocument = metadataFactory.addProcedure(GETTEXTMETADATADOCUMENT);
+        getTextMetadataDocument.setAnnotation(CouchbasePlugin.Util.getString("getTextMetadataDocument.Annotation")); //$NON-NLS-1$
+        param = metadataFactory.addProcedureParameter(KEYSPACE, TypeFacility.RUNTIME_NAMES.STRING, Type.In, getTextMetadataDocument); 
+        param.setAnnotation(CouchbasePlugin.Util.getString("getTextMetadataDocument.keyspace.Annotation")); //$NON-NLS-1$
+        param.setNullType(No_Nulls);
+        metadataFactory.addProcedureResultSetColumn(RESULT, TypeFacility.RUNTIME_NAMES.CLOB, getTextMetadataDocument); 
+        
+        Procedure getMetadataDocument = metadataFactory.addProcedure(GETMETADATADOCUMENT);
+        getMetadataDocument.setAnnotation(CouchbasePlugin.Util.getString("getMetadataDocument.Annotation")); //$NON-NLS-1$
+        param = metadataFactory.addProcedureParameter(KEYSPACE, TypeFacility.RUNTIME_NAMES.STRING, Type.In, getMetadataDocument); 
+        param.setAnnotation(CouchbasePlugin.Util.getString("getMetadataDocument.keyspace.Annotation")); //$NON-NLS-1$
+        param.setNullType(No_Nulls);
+        metadataFactory.addProcedureResultSetColumn(RESULT, TypeFacility.RUNTIME_NAMES.BLOB, getMetadataDocument);
+    }
+
+    /**
+     * All supported type in a Couchbase JSON item:
+     *   null, String, Integer, Long, Double, Boolean, 
+     *   BigInteger, BigDecimal, JsonObject, JsonArray  
+     * @param value
+     * @return
+     */
+    private String getDataType(Object value) {
+        
+        if(value == null) {
+            return TypeFacility.RUNTIME_NAMES.NULL;
+        } else if (value instanceof String) {
+            return TypeFacility.RUNTIME_NAMES.STRING;
+        } else if (value instanceof Integer) {
+            return TypeFacility.RUNTIME_NAMES.INTEGER;
+        } else if (value instanceof Long) {
+            return TypeFacility.RUNTIME_NAMES.LONG;
+        } else if (value instanceof Double) {
+            return TypeFacility.RUNTIME_NAMES.DOUBLE;
+        } else if (value instanceof Boolean) {
+            return TypeFacility.RUNTIME_NAMES.BOOLEAN;
+        } else if (value instanceof BigInteger) {
+            return TypeFacility.RUNTIME_NAMES.BIG_INTEGER;
+        } else if (value instanceof BigDecimal) {
+            return TypeFacility.RUNTIME_NAMES.BIG_DECIMAL;
+        } 
```

Comment:
The N1qlQueryResult only contains a Iterable `com.couchbase.client.java.document.json.JsonValue`, which support 12 types are
~~~
public static boolean checkType(Object item) {
        return item == null
            || item instanceof String
            || item instanceof Integer
            || item instanceof Long
            || item instanceof Double
            || item instanceof Boolean
            || item instanceof BigInteger
            || item instanceof BigDecimal
            || item instanceof JsonObject
            || item instanceof JsonArray;
    }
~~~